src/api/venue.ts

```diff
@@ -1,10 +1,19 @@
 import Bugsnag from "@bugsnag/js";
 import firebase from "firebase/app";
 
-export const setVenueLiveStatus = async (
-  venueId: string,
-  isLive: boolean
-): Promise<void | firebase.functions.HttpsCallableResult> => {
+export interface SetVenueLiveStatusProps {
+  venueId: string;
+  isLive: boolean;
+  onError?: (msg: string) => void;
+  onFinish?: () => void;
+}
+
+export const setVenueLiveStatus = async ({
+  venueId,
+  isLive,
+  onError,
+  onFinish,
+}: SetVenueLiveStatusProps): Promise<void | firebase.functions.HttpsCallableResult> => {
```

Comment:
As called out in one of my comments on the previous thread about this, we don't want to add callbacks like this, it defeats the purpose of using `Promise`-based API's. `Promise` already gives us everything we need, the only thing we needed to do was to re-throw the error within the `.catch` block after we call `Bugsnag.notify` so that downstream code can react to it and display an error to users.

Basically the commit that changed this should be reverted, and the error should be re-thrown as described above.

Reply:
Throw an error and handle it down the line, when we have a handler implemented with https://github.com/sparkletown/internal-sparkle-issues/issues/630